modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/ResolverCache.scala

```diff
@@ -175,11 +217,51 @@ object ResolverCache {
     ttl.fold(true)(ttlVal => unixSeconds - storedTstamp < ttlVal)
 
   /** Choose which result is more appropriate for being stored in cache or combine failures */
-  private def chooseResult[A](cachedResult: Option[Lookup[A]], newResult: Lookup[A]): Lookup[A] =
+  private def chooseResult[A](
+    cachedResult: Option[Lookup[A]],
+    newResult: Lookup[A],
+    newSeconds: Int
+  ): CacheResult[A] =
     (cachedResult, newResult) match {
-      case (Some(c), n) if c == n    => c
-      case (Some(Left(c)), Left(n))  => c.combine(n).asLeft
-      case (Some(Right(c)), Left(_)) => c.asRight
-      case _                         => newResult
+      case (Some(Right(c)), Right(n)) if c == n => AlreadyCached(c, newSeconds)
+      case (Some(Right(_)), Right(n))           => NewlyCached(n, newSeconds)
+      case (Some(Left(c)), Left(n))             => CacheFailures(c.combine(n))
+      case (Some(Right(c)), Left(_))            => AlreadyCached(c, newSeconds)
+      case (None, Left(n))                      => CacheFailures(n)
+      case (None, Right(n))                     => NewlyCached(n, newSeconds)
+      case (Some(Left(_)), Right(n))            => NewlyCached(n, newSeconds)
     }
+
+  /** The result of committing a lookup to the cache, carrying information about whether the cache was updated as a consequence */
+  private[resolver] sealed trait CacheResult[A] { self =>
```

Comment:
Is this trait needed? Do we need differentiate already/newly cached values? It looks like all we need is sth like `TimestampedItem` case class?